package/MDAnalysis/analysis/lineardensity.py

```diff
@@ -200,55 +251,54 @@ def _single_frame(self):
 
     def _conclude(self):
-        k = 6.022e-1  # divide by avodagro and convert from A3 to cm3
+        avogadro = 6.022e23  # unit: mol^{-1}
+        volume_conversion = 1e-24  # unit: A^3/cm^3
+        # divide result values by avodagro and convert from A3 to cm3
+        k = avogadro * volume_conversion
 
         # Average results over the number of configurations
         for dim in ['x', 'y', 'z']:
-            for key in ['pos', 'pos_std', 'char', 'char_std']:
+            for key in ['mass_density', 'mass_density_stddev',
+                        'charge_density', 'charge_density_stddev']:
                 self.results[dim][key] /= self.n_frames
             # Compute standard deviation for the error
             # For certain tests in testsuite, floating point imprecision
             # can lead to negative radicands of tiny magnitude (yielding nan).
-            # radicand_pos and radicand_char are therefore calculated first and
+            # radicand_mass and radicand_charge are therefore calculated first
             # and negative values set to 0 before the square root
             # is calculated.
-            radicand_pos = self.results[dim][
-                'pos_std'] - np.square(self.results[dim]['pos'])
-            radicand_pos[radicand_pos < 0] = 0
-            self.results[dim]['pos_std'] = np.sqrt(radicand_pos)
+            radicand_mass = self.results[dim]['mass_density_stddev'] - \
+                np.square(self.results[dim]['mass_density'])
+            radicand_mass[radicand_mass < 0] = 0
+            self.results[dim]['mass_density_stddev'] = np.sqrt(radicand_mass)
 
-            radicand_char = self.results[dim][
-                'char_std'] - np.square(self.results[dim]['char'])
-            radicand_char[radicand_char < 0] = 0
-            self.results[dim]['char_std'] = np.sqrt(radicand_char)
+            radicand_charge = self.results[dim]['charge_density_stddev'] - \
+                np.square(self.results[dim]['charge_density'])
+            radicand_charge[radicand_charge < 0] = 0
+            self.results[dim]['charge_density_stddev'] = \
+                np.sqrt(radicand_charge)
 
         for dim in ['x', 'y', 'z']:
+            # norming factor, units of mol^-1 cm^3
             norm = k * self.results[dim]['slice_volume']
             for key in self.keys:
                 self.results[dim][key] /= norm
 
     def _add_other_results(self, other):
         # For parallel analysis
-        results = self.results
         for dim in ['x', 'y', 'z']:
-            key = 'pos'
-            key_std = 'pos_std'
-            results[dim][key] += other[dim][key]
-            results[dim][key_std] += other[dim][key_std]
-
-            key = 'char'
-            key_std = 'char_std'
-            results[dim][key] += other[dim][key]
-            results[dim][key_std] += other[dim][key_std]
+            for key in self.keys:
+                self.results[dim][key] += other.results[dim][key]
```

Comment:
Had a look at the tests and this is a very weird way of parallelizing by doing it over chunks of the system rather than by frames.

This is not the standard way of doing things in MDA (or PMDA). I might advocate that we deprecate and remove this. We should just eventually implement a `_reduce` method instead in the future (as per: https://github.com/MDAnalysis/mdanalysis/issues/2542)

Reply:
I haven't seen this method used anywhere in MDA or PMDA, so it's quite possible the test I wrote does not mirror good usage. But I can add a deprecation warning here either way.

Reply:
I suspect it is the intended usage, but yeah I don't think anyone actually uses it in the wild.

Unless @PicoCentauri disagrees, please do add the deprecation warning (you can use the decorator for it): https://github.com/MDAnalysis/mdanalysis/blob/f494c98dbc1c169854399ff6815f699718d903fe/package/MDAnalysis/lib/util.py#L2296-L2302 and just raise an issue to remove it & switch to a more standard `_reduce` over frames method.

Reply:
I agree. If there is no usage in the library we can deprecate and remove it!